src/Integration/RequestIntegration.php

```diff
@@ -37,29 +37,20 @@ public function __construct(Options $options)
     /**
      * {@inheritdoc}
      */
-    public function setupOnce(): void
+    public function bindToHub(HubInterface $hub): IntegrationInterface
     {
-        Scope::addGlobalEventProcessor(function (Event $event): Event {
-            $self = Hub::getCurrent()->getIntegration(self::class);
+        Scope::addGlobalEventProcessor([$this, 'applyToEvent']);
 
-            if (!$self instanceof self) {
-                return $event;
-            }
-
-            self::applyToEvent($self, $event);
-
-            return $event;
-        });
+        return $this;
     }
 
     /**
      * Applies the information gathered by the this integration to the event.
      *
-     * @param self                        $self    The current instance of RequestIntegration
      * @param Event                       $event   The event that will be enriched with a request
      * @param ServerRequestInterface|null $request The Request that will be processed and added to the event
      */
-    public static function applyToEvent(self $self, Event $event, ?ServerRequestInterface $request = null): void
+    public function applyToEvent(Event $event, ?ServerRequestInterface $request = null): void
```

Comment:
Since this method should not be called directly from the outside, make it `private`